persist/fs/files.go

```diff
@@ -232,6 +245,32 @@ func FilesetBefore(filePathPrefix string, namespace ts.ID, shard uint32, t time.
 	return filesBefore(matched, t)
 }
 
+// DeleteInactiveFilesets deletes any filesets that are not currently owned by the namespace
+func DeleteInactiveFilesets(filePathPrefix string, namespace ts.ID, activeShards []uint32) error {
+	var toDelete []string
+	dirs := make(map[string]bool)
```

Comment:
You can make this just a `map[string]struct{}` when constructing sets.  This is what we traditionally use, so perhaps for consistency update to use that?

i.e.
```go
dirs := make(map[string]struct{})

// set
dirs[dir] = struct{}{}

// check
if _, ok := dirs[dir]; !ok {
  // ...
}
```